src/App.js

```diff
@@ -0,0 +1,198 @@
+
+  const getWeather = () => {
+    const weatherApiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${address}&appid=${WEATHER_API_KEY}`;
+    axios
+      .get(weatherApiUrl)
+      .then(function (response) {
+        console.log(response.data);
+        setCountry(getCountryByCode(response.data.sys.country));
+        setLocationInfo(response.data);
+      })
+      .catch(function (error) {
+        setError(true);
+      });
+  };
+
+  const kelvinToFahrenheit = (k) => {
+    return (k - 273.15).toFixed(2);
+  };
+
+  const getLocation = () => {
+    if (!navigator.geolocation) {
+      setStatus("Geolocation is not supported by your browser");
+    } else {
+      setStatus("Locating...");
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          setStatus(null);
+          setLat(position.coords.latitude);
+          setLng(position.coords.longitude);
+        },
+        () => {
+          setStatus("Unable to retrieve your location");
+        }
+      );
+    }
+  };
+
+  useEffect(() => {
+    if (lat && lng) {
+      getLocationLatLong(lat, lng);
+    }
+  }, [lat, lng]);
+
+  useEffect(() => {
+    if (address) {
+      getWeather();
+    }
+  }, [address]);
+
+  const getLocationLatLong = (lat, lang) => {
+    Geocode.fromLatLng(lat, lang).then(
+      (response) => {
+        const address = response.results[0].formatted_address;
+        let city, state, country;
+        for (
+          let i = 0;
+          i < response.results[0].address_components.length;
+          i++
+        ) {
+          for (
+            let j = 0;
+            j < response.results[0].address_components[i].types.length;
+            j++
+          ) {
+            switch (response.results[0].address_components[i].types[j]) {
+              case "locality":
+                city = response.results[0].address_components[i].long_name;
+                break;
+              case "administrative_area_level_1":
+                state = response.results[0].address_components[i].long_name;
+                break;
+              case "country":
+                country = response.results[0].address_components[i].long_name;
+                break;
+            }
+          }
+        }
+        setAddress(city);
+      },
+      (error) => {
+        console.error(error);
+      }
+    );
+  };
+
+  return (
+    <div className="App">
+      <header className="App-header pt-5">
+        <div className="container">
+          <div className="wrapper pt-5">
+            <div className="row">
+              <div className="col-2"></div>
+              <div className="col-8">
+                <div className="input-group">
+                  <FaCrosshairs
+                    onClick={getLocation}
+                    className="icon-location"
+                  />
+                  <input
+                    type="text"
+                    className="form-control"
+                    placeholder="Enter Location..."
```

Comment:
formatting one line

Reply:
@Henshall when line length increases from 80 characters then readability gets compromised. To improve readability single line is separated into multiple lines. see industry standard code editor extension https://prettier.io/